transport/grpc/dial.go

```diff
@@ -400,7 +400,13 @@ func addOpenTelemetryStatsHandler(opts []grpc.DialOption, settings *internal.Dia
 	if settings.TelemetryDisabled {
 		return opts
 	}
-	return append(opts, grpc.WithStatsHandler(otelGRPCStatsHandler()))
+	otelOpts := []otelgrpc.Option{}
+	for _, opt := range settings.OpenTelemetryOpts {
+		if opt, ok := opt.(otelgrpc.Option); ok {
```

Comment:
I think we should return an error when typecasting to both `otelgrpc.Option` and `otelhttp.Option` fails. That way users will understand immediately (at client init) that they have passed an invalid type. This can probably be done in `DialSettings.Validate`.  You can similarly handle the conversions in new accessors: `DialSettings.OtelGRPCOpts` and `DialSettings.OtelHTTPOpts`, in order to keep this logic out of `transport`.

Reply:
Done here https://github.com/googleapis/google-api-go-client/pull/3130/commits/899af8484d6587931824d127259c0bd9580ecfca